gc3libs/workflow.py

```diff
@@ -401,7 +401,10 @@ def redo(self, from_stage=0, *args, **kwargs):
             task.redo(*args, **kwargs)
         # All other tasks should be put in NEW again
         for i in range(from_stage+1, len(self.tasks)):
-            self.tasks[i].execution.state = Run.State.NEW
+            try:
+                self.tasks[i].redo(*args, **kwargs)
+            except IndexError:
```

Comment:
The assertion only fails for a particular custom implementation of SequentialTaskCollection:

File "/Users/mdh/gc3pie/gc3libs/core.py", line 1694, in redo
    task.redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 401, in redo
    task.redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 401, in redo
    task.redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 405, in redo
    self.tasks[i].redo(*args, **kwargs)
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 399, in redo
    task = self.stage()
  File "/Users/mdh/gc3pie/gc3libs/workflow.py", line 438, in stage
    assert (0 <= self._current_task < len(self.tasks)), ("Invalid value self._current_task=%d in %s" % (self._current_task, self))
AssertionError: Invalid value self._current_task=0 in <MultiRunJobCollection(name='illuminati_run', n=0, submission_id=144)>